pom.xml

```diff
@@ -364,6 +364,31 @@
                     <target>${maven.compiler.target}</target>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>4.2.1</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Export-Package>
+                            com.github.davidmoten.rtree;
+                            com.github.davidmoten.rtree.geometry;
+                            com.github.davidmoten.rtree.internal;
```

Comment:
```suggestion
                            com.github.davidmoten.rtree;
                            com.github.davidmoten.rtree.geometry;
                            com.github.davidmoten.rtree.internal;
```

Reply:
I still think it should be 

```
!com.github.davidmoten.rtree.internal.*;
com.github.davidmoten.rtree.*;
```
This matches the default behaviour of Felix plugin. From https://felix.apache.org/documentation/subprojects/apache-felix-maven-bundle-plugin-bnd.html#default-behavior:

>&lt;Export-Package&gt; is now assumed to be the set of packages in your local Java sources, excluding the default package '.' and any packages containing 'impl' or 'internal'. (before version 2 of the bundleplugin it was based on the symbolic name)

The internal package classes will still be available at runtime for use by the exported classes (but presumably not available in an OSGi context for direct instantiation). 

If you think any classes have been labelled internal yet they are exposed in the public API then let me know and I'll have a look with a view to moving them around.

Reply:
I think I'm convinced. I've extended the example with a usage of a class which has a reference to an internal class. Eg. when ```Entries.entry(object, geometry)``` is called, an instance of EntryDefault created. Both bundles may run on the same jvm only the visibility is restricted.

Reply:
Thanks for trying it out. So I assume that you can't call `EntryDefault.entry(...)` for instance?

Reply:
Yes, I can't. The example bundle is not activated due to missing requirement.